demos/faults.cc

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under both the 3-Clause BSD License and the GPLv2, found in the
+ * LICENSE and LICENSE.GPL-2.0 files, respectively, in the root directory.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause OR GPL-2.0
+ */
+
+#include "faults.h"
+
+#include <setjmp.h>
+
+#include <cstring>
+
+namespace {
+
+// The fault handler that should be called from the signal handler, and the
+// sigjmp context that should be jumped to after that.
+//
+// Declaring these thread-local might be a bit aspirational -- they definitely
+// *shouldn't* be shared across threads, but the current implementation doesn't
+// necessarily guarantee it's safe to use RunWithFaultHandler on two threads
+// without external synchronization.
+thread_local PosixFaultHandler fault_handler;
+thread_local sigjmp_buf signal_handler_jmpbuf;
+
+void SignalHandler(int signal, siginfo_t *info, void *ucontext) {
```

Comment:
Ah, I think this should be `extern "C"` per https://en.cppreference.com/w/cpp/utility/program/signal "Signal handlers are expected to have C linkage and, in general, only use the features from the common subset of C and C++. It is implementation-defined if a function with C++ linkage can be used as a signal handler."

Reply:
Changed.